qutip/tests/test_entropy.py

```diff
@@ -89,6 +89,56 @@ def test_component_selection(self):
         assert abs(qutip.entropy_mutual(dm, [0, 2], [1]) - expect) < 1e-13
 
 
+class TestRelativeEntropy:
+    def test_rho_or_sigma_not_oper(self):
+        rho = qutip.ket("00")
+        sigma = qutip.ket("01")
+        with pytest.raises(TypeError) as exc:
+            qutip.entropy_relative(qutip.ket2dm(rho), sigma)
+        assert str(exc.value) == "Inputs must be density matrices."
```

Comment:
minor nit, and maybe it's just because I learned Python later than other stuff, but `exc` is logically of out-of-scope in the assertion.  I'm pretty sure `pytest.raises` takes a message argument anyway, so you can put the test in there.

Reply:
`pytest.raises` only takes a regular expression pattern via the `match` argument, which results in the equivalent code being something like:
```with pytest.raises(TypeError, match=re.escape("Inputs must be density matrices."))
```
which is not particularly pretty.

I tend to use this form since it easily allows checking other attributes of the exception (not that there are any here).

Reply:
Fair enough, then.  The scoping thing is probably me just being a bit weird - looking at the pytest documentation, they actually have this kind of thing as a direct example.